packages/react-components/react-overflow/src/components/Overflow.tsx

```diff
@@ -24,21 +30,27 @@ export const Overflow = React.forwardRef((props: OverflowProps, ref) => {
 
   const { children, minimumVisible, overflowAxis = 'horizontal', overflowDirection, padding } = props;
 
-  const [hasOverflow, setHasOverflow] = React.useState(false);
-  const [itemVisibility, setItemVisibility] = React.useState<Record<string, boolean>>({});
-  const [groupVisibility, setGroupVisibility] = React.useState<Record<string, OverflowGroupState>>({});
+  const [overflowState, setOverflowState] = React.useState<OverflowState>({
+    hasOverflow: false,
+    itemVisibility: {},
+    groupVisibility: {},
+  });
 
   // useOverflowContainer wraps this method in a useEventCallback.
-  // TODO: Do we need a useEventCallback here too?
   const update: OnUpdateOverflow = data => {
-    setHasOverflow(() => data.invisibleItems.length > 0);
-    setItemVisibility(() => {
-      const newState: Record<string, boolean> = {};
```

Comment:
These seperate states were always updated togeter, it makes no sense to have them separate